### PR TITLE
CDD-2428: re-create ttl index when it's value changes

### PR DESCRIPTION
**Pull request by paulblakehmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/apinotificationqueue/repository/NotificationRepository.scala [hunk 1]
```diff
@@ -197,4 +204,19 @@ class NotificationMongoRepository @Inject()(mongoDbProvider: MongoDbProvider,
       .cursor
       .collect[List](-1, reactivemongo.api.Cursor.FailOnError[List[NotificationWithIdAndPulled]]())
   }
+
+  private def dropInvalidIndexes: Future[_] =
+    collection.indexesManager.list().flatMap { indexes =>
+      indexes
+        .find { index =>
+          index.name.contains(ttlIndexName) &&
+            !index.options.getAs[Int]("expireAfterSeconds").contains(config.ttlInSeconds)
```
Comment (rashadmughal) [inline app/uk/gov/hmrc/apinotificationqueue/repository/NotificationRepository.scala]: extract "expireAfterSeconds" as a constant?

File: app/uk/gov/hmrc/apinotificationqueue/repository/NotificationRepository.scala [hunk 2]
```diff
@@ -64,6 +64,19 @@ class NotificationMongoRepository @Inject()(mongoDbProvider: MongoDbProvider,
 
   private implicit val format: Format[ClientNotification] = ClientNotificationJF
 
+  private val ttlIndexName = "dateReceived-Index"
+  
+  private val ttlIndex = Index(
+    key = Seq("notification.dateReceived" -> IndexType.Descending),
```
Comment (rashadmughal) [inline app/uk/gov/hmrc/apinotificationqueue/repository/NotificationRepository.scala]: extract notification.dateReceived too

File: app/uk/gov/hmrc/apinotificationqueue/repository/NotificationRepository.scala (hunk 1 above)
Comment (paulblakehmrc) [reply app/uk/gov/hmrc/apinotificationqueue/repository/NotificationRepository.scala]: Done

File: app/uk/gov/hmrc/apinotificationqueue/repository/NotificationRepository.scala (hunk 2 above)
Comment (paulblakehmrc) [reply app/uk/gov/hmrc/apinotificationqueue/repository/NotificationRepository.scala]: Agreed not to do.

**Pull request merged**